tools/apicoverage/apicoverage.go

```diff
@@ -187,58 +186,69 @@ func getRelevantLogs(fields []string) *string {
 	return nil
 }
 
-func createCases(tcName string, covered map[string]int, notCovered map[string]int) []testgrid.TestCase {
-	var tc []testgrid.TestCase
-
+func addTestCases(tcName string, covered map[string]int, notCovered map[string]int, testSuite *junit.TestSuite) {
 	if (len(covered) == 0 && len(notCovered) == 0) {
-		return tc
+		return
 	}
 
 	var percentCovered = float32(100 * len(covered) / (len(covered) + len(notCovered)))
-	tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: percentCovered}}
-	tc = append(tc, testgrid.TestCase{Name: tcName, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+	testCase := junit.TestCase{Name: tcName}
+	testCase.AddProperty(apiCoverage, fmt.Sprintf("%f", percentCovered))
+	testSuite.TestCases = append(testSuite.TestCases, testCase)
 
 	for key, value := range covered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+		coveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		coveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		testSuite.TestCases = append(testSuite.TestCases, coveredTestCase)
 	}
 
 	for key, value := range notCovered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: true})
+		notCoveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		notCoveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		failureMessage := "failure"
+		notCoveredTestCase.Failure = &failureMessage
+		testSuite.TestCases = append(testSuite.TestCases, notCoveredTestCase)
 	}
-	return tc
 }
 
 func createTestgridXML(coverage *OverallAPICoverage, artifactsDir string) {
-	tc := createCases(overallRoute, coverage.RouteAPICovered, coverage.RouteAPINotCovered)
-	tc = append(tc, createCases(overallConfig, coverage.ConfigurationAPICovered, coverage.ConfigurationAPINotCovered)...)
-	tc = append(tc, createCases(overallService, coverage.ServiceAPICovered, coverage.ServiceAPINotCovered)...)
-	ts := testgrid.TestSuite{TestCases: tc}
+	testSuite := junit.TestSuite{Name: ""}
+	addTestCases(overallRoute, coverage.RouteAPICovered, coverage.RouteAPINotCovered, &testSuite)
+	addTestCases(overallConfig, coverage.ConfigurationAPICovered, coverage.ConfigurationAPINotCovered, &testSuite)
+	addTestCases(overallService, coverage.ServiceAPICovered, coverage.ServiceAPINotCovered, &testSuite)
+
+	testSuites := junit.TestSuites{}
+	if err := testSuites.AddTestSuite(&testSuite); nil != err {
+		log.Fatalf("Cannot add testSuite to testSuites: %v", err)
```

Comment:
As I mentioned in the function, this will never happen because you never return an error.

Reply:
this function can return error now as mentioned above, however in this case it won't, removed anyway